src/lib/core/utils/data-element-constraints.ts

```diff
@@ -71,6 +73,10 @@ function variableConstraintPredicate(
       constraint.allowedShapes.includes(variable.dataShape)) &&
       (constraint.allowedTypes == null ||
         constraint.allowedTypes.includes(variable.type)) &&
+      (constraint.maxNumValues == null ||
+        //this case should be category anyway, but is that a strong enough thing to assume here?
+        variable.distinctValuesCount === undefined ||
```

Comment:
This can also be removed, due to the type narrowing, if the changes in `src/lib/core/types/study.ts` are undone.